crates/biome_graphql_analyze/src/lint/nursery/use_deprecated_date.rs

```diff
@@ -0,0 +1,139 @@
+    type Signals = Option<Self::State>;
+    type Options = UseDeprecatedDateOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+        let name = node.name().ok()?;
+
+        if name.to_trimmed_string() != "deprecated" {
+            return None;
+        }
+
+        let Some(arguments) = node.arguments() else {
+            return Some((DeprecatedDateIssue::Missing, node.clone()));
+        };
+        let arguments = arguments.arguments();
+        let argument_name = ctx.options().argument_name.clone();
+        let Some(argument) = arguments.into_iter().find(|argument| {
+            argument
+                .name()
+                .is_ok_and(|name| name.to_trimmed_string() == argument_name)
+        }) else {
+            return Some((DeprecatedDateIssue::Missing, node.clone()));
+        };
+
+        let argument_value = argument.value().ok()?;
+
+        let Some(argument_string_value) = argument_value.as_graphql_string_value() else {
+            return Some((DeprecatedDateIssue::Invalid, node.clone()));
+        };
+
+        let due_date_value = argument_string_value.inner_string_text().ok()?;
+        let due_date = due_date_value.to_string().parse();
+
+        if due_date.is_err() {
+            return Some((DeprecatedDateIssue::Invalid, node.clone()));
+        }
+
+        let now = Zoned::now();
+
+        if now.date() > due_date.ok()? {
+            return Some((DeprecatedDateIssue::Due, node.clone()));
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let span = ctx.query().range();
+        let argument_name = ctx.options().argument_name.clone();
+
+        match state.0 {
+            DeprecatedDateIssue::Missing => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    span,
+                    markup! {
+                        "The directive `@deprecated` should have a `"{argument_name}"` argument."
+                    },
+                )
+                .note(markup! {
+                        "Add a `"{argument_name}"` argument to the directive."
+                }),
+            ),
+            DeprecatedDateIssue::Invalid => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    span,
+                    markup! {
+                        "The `"{argument_name}"` argument in the directive `@deprecated` has an invalid value."
+                    },
+                )
+                .note(markup! {
+                    "The argument must match the `YYYY-MM-DD` format"
+                }),
+            ),
+            DeprecatedDateIssue::Due => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    span,
+                    markup! {
+                        "The deprecation has passed it's due date"
```

Comment:
```suggestion
                        "The deprecation has passed its due date"
```

That's the possessive